File: frontend/src/types/employee.ts
import { Internship } from './internship'

export type Employee = {
  id: number
  firstName: string
  lastName: string
  title: string
  email: string
  companyId: number
  companyName: string
}

export type GetEmployeeByIdViewModel = {
  id: number
  firstName: string
  lastName: string
  title: string
  email: string
  companyId: number
  companyName: string
  internships: Internship[]
}

File: backend/Internships/Internships.Application/Entities/Employee.cs
using System.Collections.Generic;

namespace Internships.Core.Entities
{
    public class Employee : AuditableBaseEntity
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Title { get; set; }
        public string Email { get; set; }

        public int CompanyId { get; set; }
        public virtual Company Company { get; set; }
        
        public virtual ICollection<Internship> Internships { get; set; }
        public bool IsEnabled { get; set; }
    }
}